pkg/kubernetes/version_test.go

```diff
@@ -0,0 +1,348 @@
+								"labels": map[string]interface{}{
+									"label1": "test-label1",
+								},
+								"uid": "cec15437-4e6a-11ea-9788-4201ac100006",
+							},
+						},
+					}
+					isVersioned = kc.IsVersioned(&fakeResource)
+				})
+				It("returns true", func() {
+					Expect(isVersioned).To(Equal(true))
+				})
+			})
+			When("the resource kind is statefulSet", func() {
+				BeforeEach(func() {
+					fakeResource := unstructured.Unstructured{
+						Object: map[string]interface{}{
+							"kind": "statefulSet",
+							"metadata": map[string]interface{}{
+								"name":              "fakeName",
+								"namespace":         "test-namespace2",
+								"creationTimestamp": "2020-02-13T14:12:03Z",
+								"labels": map[string]interface{}{
+									"label1": "test-label1",
+								},
+								"uid": "cec15437-4e6a-11ea-9788-4201ac100006",
+							},
+						},
+					}
+					isVersioned = kc.IsVersioned(&fakeResource)
+				})
+				It("returns false", func() {
+					Expect(isVersioned).To(Equal(false))
+				})
+			})
+		})
+	})
+	Context("#IncrementVersion", func() {
+		When("current version is 1", func() {
+			BeforeEach(func() {
+				kc = kubernetes.NewController()
+				updatedVersion = kc.IncrementVersion("1")
+				expectedVersion = kubernetes.SpinnakerVersion{
+					Long:  "v002",
+					Short: "2",
+				}
+			})
+			It("returns expected version", func() {
+				Expect(updatedVersion).To(Equal(expectedVersion))
+			})
+		})
+		When("current version is 999", func() {
+			BeforeEach(func() {
+				kc = kubernetes.NewController()
+				updatedVersion = kc.IncrementVersion("999")
+				expectedVersion = kubernetes.SpinnakerVersion{
+					Long:  "v000",
+					Short: "0",
+				}
+			})
+			It("returns expected version", func() {
+				Expect(updatedVersion).To(Equal(expectedVersion))
+			})
+		})
+	})
+	Context("#AddSpinnakerVersionAnnotations", func() {
+		When("kind is a replicaset", func() {
+			BeforeEach(func() {
+				fakeResource = unstructured.Unstructured{
+					Object: map[string]interface{}{
+						"kind": "test-kind",
+						"metadata": map[string]interface{}{
+							"name":              "test-name",
+							"namespace":         "test-namespace2",
+							"creationTimestamp": "2020-02-13T14:12:03Z",
+							"labels": map[string]interface{}{
+								"label1": "test-label1",
+							},
+							"annotations": map[string]interface{}{
+								"strategy.spinnaker.io/versioned": "true",
+								"moniker.spinnaker.io/cluster":    "test-kind test-name",
+							},
+							"uid": "cec15437-4e6a-11ea-9788-4201ac100006",
+						},
+					},
+				}
+
+				fakeVersion = kubernetes.SpinnakerVersion{
+					Long:  "v002",
+					Short: "2",
+				}
+
+				kc = kubernetes.NewController()
+				err = kc.AddSpinnakerVersionAnnotations(&fakeResource, fakeVersion)
+
+			})
+			It("expect error not to have occured", func() {
+				Expect(err).To(BeNil())
+			})
+		})
+	})
+
+	Context("#AddSpinnakerVersionLabels", func() {
```

Comment:
Move this Context to the `label_test.go` file.

Reply:
done